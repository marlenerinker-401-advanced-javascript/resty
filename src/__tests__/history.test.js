import React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route } from 'react-router-dom';

import History from '../history.js';
const testFunction = jest.fn();

describe('History tests', () => {
  it('should render at application start', () => {
    let app = mount(<BrowserRouter><History archive={[]} populate={testFunction}/></BrowserRouter>);
    expect(app.find('p').exists()).toBe(true);
    expect(app.find('li').exists()).toBe(false);

  });

  it('should render with data ', () => {
    let app = mount(<BrowserRouter><History archive={[{name: 'test'}, {name: 'test2'}]} populate={testFunction}/></BrowserRouter>);
    expect(app.find('p').exists()).toBe(true);
    expect(app.find('li').exists()).toBe(true);
  });


  it('renders correctly', () => {
    const DOM = renderer.create(<BrowserRouter><History archive={[{name: 'test'}, {name: 'test2'}]} populate={testFunction}/></BrowserRouter>).toJSON();
    expect(DOM).toMatchSnapshot();
  });

});
