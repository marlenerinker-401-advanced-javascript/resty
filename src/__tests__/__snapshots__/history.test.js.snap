// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History tests renders correctly 1`] = `
<div
  id="history"
>
  <p
    id="intro"
  >
    This is your API call history. Click on a call to see details.
  </p>
  <section
    id="api-calls"
  >
    <ul>
      <li
        id={0}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<pre class=\\"__json-pretty-error__\\">undefined undefined</pre>",
            }
          }
          id="0call"
          onClick={[Function]}
        />
      </li>
      <li
        id={1}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<pre class=\\"__json-pretty-error__\\">undefined undefined</pre>",
            }
          }
          id="1call"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
</div>
`;
